plugins/http_plugin/http_plugin.cpp

```diff
@@ -287,13 +328,201 @@ namespace eosio {
+            :object(std::move(object))
+            ,impl(impl)
+            {
+               count = detail::in_flight_sizeof(object);
+               impl.bytes_in_flight += count;
+            }
+
+            ~in_flight() {
+               if (count) {
+                  impl.bytes_in_flight -= count;
+               }
+            }
+
+            // No copy constructor, but allow move
+            in_flight(const in_flight&) = delete;
+            in_flight(in_flight&& from)
+            :object(std::move(from.object))
+            ,count(from.count)
+            ,impl(from.impl)
+            {
+               from.count = 0;
+            }
+
+            // No copy assignment, but allow move
+            in_flight& operator=(const in_flight&) = delete;
+            in_flight& operator=(in_flight&& from) {
+               object = std::move(from.object);
+               count = from.count;
+               impl = from.impl;
+               from.count = 0;
+            }
+
+            /**
+             * const accessor
+             * @return const reference to the contained object
+             */
+            const T& operator* () const {
+               return object;
+            }
+
+            /**
+             * mutable accessor (can be moved frmo)
+             * @return mutable reference to the contained object
+             */
+            T& operator* () {
+               return object;
+            }
+
+            T object;
+            size_t count;
+            http_plugin_impl& impl;
+         };
+
+         /**
+          * convenient wrapper to make an in_flight<T>
+          */
+         template<typename T>
+         static auto make_in_flight(T&& object, http_plugin_impl& impl) {
+            return in_flight<T>(std::forward<T>(object), impl);
+         }
+
+         /**
+          * Make an internal_url_handler that will run the url_handler on the app() thread and then
+          * return to the http thread pool for response processing
+          *
+          * @pre b.size() has been added to bytes_in_flight by caller
+          * @param priority - priority to post to the app thread at
+          * @param next - the next handler for responses
+          * @return the constructed internal_url_handler
+          */
+         detail::internal_url_handler make_app_thread_url_handler( int priority, url_handler next ) {
+            return [this, priority, next=std::move(next)]( detail::abstract_conn_ptr conn, string r, string b, url_response_callback then ) mutable {
+               auto tracked_b = make_in_flight(std::move(b), *this);
+               if (!conn->verify_max_bytes_in_flight()) {
+                  return;
+               }
+
+               // post to the app thread capturing the body_scope
+               app().post( priority, [next=std::move(next), conn=std::move(conn), r=std::move(r), tracked_b=std::move(tracked_b), then=std::move(then)]() mutable {
```

Comment:
Do not move `next` as it is captured by this lambda that is used multiple times. Also add comment why it is not being moved and update existing comment as it is not clear what is meant by `body_scope`.